### PR TITLE
[docs] Update composition errors docs

### PR DESCRIPTION
**Pull request by smyrick**
Remove the reference to Apollo Gateway and add link to composition hints

### CODE REVIEW

File: docs/source/errors.mdx [hunk 1]
```diff
@@ -4,13 +4,18 @@ description: Learn the error codes you can receive if composition fails.
 toc: false
 ---
 
-When Apollo Gateway attempts to compose the schemas provided by your [subgraphs](./building-supergraphs/subgraphs-overview/) into a supergraph schema, it confirms that:
+When Apollo Federation attempts to compose the schemas provided by your [subgraphs](./building-supergraphs/subgraphs-overview/) into a supergraph schema, it confirms that:
```
Comment (Meschreiber) [inline docs/source/errors.mdx:7]:
```suggestion
When GraphOS attempts to compose the schemas provided by your [subgraphs](./building-supergraphs/subgraphs-overview/) into a supergraph schema, it confirms that:
```

Comment (Meschreiber) [reply docs/source/errors.mdx:7]: As I understand it, Apollo Federation is the architecture, not the thing doing composition. Would GraphOS work here?

File: docs/source/errors.mdx [hunk 2]
```diff
@@ -4,13 +4,18 @@ description: Learn the error codes you can receive if composition fails.
...
 
-- The subgraphs are valid
+- The subgraphs have valid GraphQL and Federation syntax
 - The resulting supergraph schema is valid
-- The gateway has all of the information it needs to execute operations against the resulting schema
+- The router/gateway has all of the information it needs to execute operations against the resulting schema
```
Comment (Meschreiber) [inline docs/source/errors.mdx:11]:
```suggestion
- The graph router or gateway has all of the information it needs to execute operations against the resulting schema
```

File: docs/source/errors.mdx [hunk 3]
```diff
@@ -4,13 +4,18 @@ description: Learn the error codes you can receive if composition fails.
...
 
-If Apollo Gateway encounters an error, composition fails. This document lists subgraph validation and composition error codes, along with their root causes.
+If composition encounters an error, it fails to create the supergraph and will not pick up any subgraph changes. This document lists subgraph validation and composition error codes, along with their root causes.
```
Comment (Meschreiber) [inline docs/source/errors.mdx:13]:
```suggestion
If composition fails, a new supergraph schema cannot be generated. This document lists subgraph validation and composition error codes, along with their root causes.
```

File: docs/source/errors.mdx [hunk 4]
```diff
@@ -4,13 +4,18 @@ description: Learn the error codes you can receive if composition fails.
...
+
+<Note>
+Composition might find a schema that technically valid but it can flag potential improvements or hints. These will not throw an error but will be logged as part of the composition build step.
+To learn more, see our docs on [composition hints](/federation/hints/).
```
Comment (Meschreiber) [inline docs/source/errors.mdx:17]:
```suggestion

Composition may flag potential improvements or _hints_ on schemas which are technically valid. These hints aren't  errors, and are instead logged as part of the composition build step.
To learn more, see the [composition hints doc](/federation/hints/).
```

**Review by Meschreiber — APPROVED** [review]
Approved with some suggestions for clarity.

File: docs/source/errors.mdx (hunk 1 above)
Comment (smyrick) [reply docs/source/errors.mdx:7]: Yea, this works better to recognize the entire ecosystem

**Pull request merged**